satpy/readers/ahi_hsd.py

```diff
@@ -620,9 +619,14 @@ def _read_data(self, fp_, header):
         """Read data block."""
         nlines = int(header["block2"]['number_of_lines'][0])
         ncols = int(header["block2"]['number_of_columns'][0])
+        chunks = da.core.normalize_chunks("auto",
+                                          shape=(nlines, ncols),
+                                          limit=get_chunk_size_limit(),
+                                          dtype='f8',
+                                          previous_chunks=(550, 550))
```

Comment:
In your testing does the `550, 550` previous chunks mean:

1. It preserves the square shape of the chunks
2. It tries to preserve 550 as a building block (factor) when building larger chunks
3. Both 1 and 2

Also, what if we request a chunk size even smaller? How does the 550 play a part?

Reply:
I meant is as 2.

If the chunk size is smaller, then I think 550 is just ignored...

Reply:
In that case do we want the previous chunks to be even smaller? What "rule" can reader developers use to come up with this in the future?

Reply:
I'm not sure. I based 550 on the extensive usage by @simonrp84 with this data, but maybe there is a smarter way. In particular, I just saw that jp01 data (as opposed to fldk) does not have sizes multiple of 550...
In fact, the data in hsd is store as one big array, so there is no inherent chunking needed in principle. So just letting dask decide of the chunk size should be good enough, but this need further performance testing I think.

@simonrp84 do you have observations you want to report on this topic?

Reply:
To sum up some research, leaving out the previous chunk size allow for much more fine grained changes in the chunk sizes. I could also not see any significant performance difference when the chunk size are a multiple of 550 or not in my tests, but I will let @simonrp84 have the last word on this.

Reply:
Just to reiterate the reason AHI had questionable chunking in the first place: the old default of 1024 would leave and odd small chunk at the end of each segment which from memory were 2200 rows. This was made worse when things were concatenated and then used in caching where zarr wanted (I think) all chunks to be the same size except the last one was allowed to be smaller. We also want to avoid the case of `old_arr.rechunk(old_arr.chunks[0][0])` where that first chunk might be 1024, but the others might not be and doing this rechunk would be costly and create chunks consisting of parts from different segment files.

I think this 550 is fine. If testing shows that `normalize_chunks` without `previous_chunks` can give us something consistent and aligned well with the segment shape then I'm also OK removing the previous chunks.

Reply:
removing `previous_chunks`will not give use well aligned segment shapes afaics.

Reply:
Ok. I guess 550 is good enough with me.

Reply:
alright, @simonrp84 just reports on slack that it doesn't seem to have much impact, so I'm leaning towards removing `previous_chunks`

Reply:
Agreed, recent version of satpy + dask don't seem sensitive to the chunk size so I'm fine for this to be removed.